Smart Contract

```diff
@@ -70,37 +69,36 @@ contract Auction {
     function Auction ();
     
     function createNewAuction (uint _auctionTime, uint _minimumBid ) returns (uint) {
-        uint AuctionID = NumberOfAuctions.length;
-        NumberOfAuctions.push(1);
+        uint AuctionID = numberOfAuctions;
+        numberOfAuctions = numberOfAuctions + 1;
         endAuctionTime[AuctionID] = now + _auctionTime;
         minimumBid[AuctionID] = _minimumBid;
         auctionEnded[AuctionID] = false; 
         seller[AuctionID] = msg.sender;
         return AuctionID;
```

Comment:
Maybe have an "AuctionCreated" event? This could be very useful for a front end :~) (P.s. I have been working on JavaScript hardcore. Get ready for some front end next year.)